pkg/cmd/connector/update/update.go

```diff
@@ -2,118 +2,177 @@ package update
 
 import (
 	"encoding/json"
-	"io/ioutil"
+	"fmt"
+	"github.com/jackdelahunt/survey-json-schema/pkg/surveyjson"
+	"github.com/redhat-developer/app-services-cli/pkg/core/cmdutil/flagutil"
+	"github.com/redhat-developer/app-services-cli/pkg/shared/connection/api"
+	"github.com/redhat-developer/app-services-cli/pkg/shared/connectorutil"
+	"github.com/spf13/cobra"
 	"os"
+	// embed static HTML file
+	_ "embed"
 
-	"github.com/redhat-developer/app-services-cli/pkg/cmd/connector/connectorcmdutil"
-	"github.com/redhat-developer/app-services-cli/pkg/cmd/registry/artifact/util"
-	"github.com/redhat-developer/app-services-cli/pkg/core/cmdutil/flagutil"
 	"github.com/redhat-developer/app-services-cli/pkg/core/ioutil/dump"
+	"github.com/redhat-developer/app-services-cli/pkg/core/localize"
+	connectorerror "github.com/redhat-developer/app-services-sdk-go/connectormgmt/apiv1/error"
+	"gopkg.in/AlecAivazis/survey.v1"
 
-	"github.com/redhat-developer/app-services-cli/pkg/shared/connection"
 	"github.com/redhat-developer/app-services-cli/pkg/shared/factory"
-	"github.com/spf13/cobra"
+	connectormgmtclient "github.com/redhat-developer/app-services-sdk-go/connectormgmt/apiv1/client"
 )
 
 type options struct {
-	file string
-
+	name         string
 	outputFormat string
 	f            *factory.Factory
-	id           string
 }
 
-// NewUpdateCommand creates a new command to update a connector
 func NewUpdateCommand(f *factory.Factory) *cobra.Command {
-
 	opts := &options{
 		f: f,
 	}
 
 	cmd := &cobra.Command{
 		Use:     "update",
-		Short:   f.Localizer.MustLocalize("connector.update.cmd.shortDescription"),
-		Long:    f.Localizer.MustLocalize("connector.update.cmd.longDescription"),
-		Example: f.Localizer.MustLocalize("connector.update.cmd.example"),
-		Hidden:  true,
+		Short:   f.Localizer.MustLocalize("connector.create.cmd.shortDescription"),
+		Long:    f.Localizer.MustLocalize("connector.create.cmd.longDescription"),
+		Example: f.Localizer.MustLocalize("connector.create.cmd.example"),
 		Args:    cobra.NoArgs,
 		RunE: func(cmd *cobra.Command, args []string) error {
-
 			validOutputFormats := flagutil.ValidOutputFormats
 			if opts.outputFormat != "" && !flagutil.IsValidInput(opts.outputFormat, validOutputFormats...) {
 				return flagutil.InvalidValueError("output", opts.outputFormat, validOutputFormats...)
 			}
 
-			return runUpdateCommand(opts)
+			return runCreate(opts)
 		},
 	}
-
-	flags := connectorcmdutil.NewFlagSet(cmd, f)
-	flags.StringVar(&opts.file, "file", "", f.Localizer.MustLocalize("connector.file.flag.description"))
```

Comment:
Would we allow users to edit connector spec file and run update on this?

Reply:
My idea of the connector update was that it is always pointing to a connector that is created already and you want to update it. Updating the json build file I think is a different idea that shouldn't be covered by the connector update command

Reply:
> My idea of the connector update was that it is always pointing to a connector that is created already and you want to update it. 

That would be also my understanding of updating item. 

>  Updating the json build file I think is a different idea t

I did not implied that would be new file. I meant that users can stream connector changes and patch it:

- Sending patch request 
- Sending whole file that user edited. etc.

all of that can be done by streaming (reading standard input) in update command. 

This would be alternative way to handle updates etc.